spec/lib/fastly_nsq/message_queue/consumer_spec.rb

```diff
@@ -17,49 +14,47 @@ def fake_consumer
     end
   end
 
-  describe 'when using the real queue', fake_queue: false do
-    before(:example) do
-      @fake_consumer = fake_consumer
-      allow(Nsq::Consumer).to receive(:new).and_return(@fake_consumer)
+  describe 'when connector connects to a backend Consumer' do
+    let(:consumer) do
+      MessageQueue::Consumer.new topic: topic, channel: channel, connector: ->(_) { backend }
     end
 
-    it 'forwards #pop to Nsq::Consumer' do
+    it 'forwards #pop' do
+      expect(backend).to receive(:pop)
       consumer.pop
-      expect(@fake_consumer).to have_received(:pop)
     end
 
-    it 'forwards #size to Nsq::Consumer' do
+    it 'forwards #pop_without_blocking' do
+      expect(backend).to receive(:pop_without_blocking)
+      consumer.pop_without_blocking
```

Comment:
You can use shoulda-matchers here if you like:

```ruby
it { is_expected.to delegate_method(:pop_without_blocking).to(:backend) }
```

http://www.rubydoc.info/github/thoughtbot/shoulda-matchers/Shoulda%2FMatchers%2FIndependent%3Adelegate_method